lib/docker_sync/sync_strategy/unison.rb

```diff
@@ -82,6 +85,7 @@ def start_container
           if exists == ''
             say_status 'ok', "creating #{container_name} container", :white if @options['verbose']
             cmd = "docker run -p '#{@options['sync_host_port']}:#{UNISON_CONTAINER_PORT}' -v #{volume_name}:#{@options['dest']} -e UNISON_VERSION=#{UNISON_VERSION} -e UNISON_WORKING_DIR=#{@options['dest']} --name #{container_name} -d #{UNISON_IMAGE}"
+            cmd = cmd + " && docker exec #{container_name} chown #{Process.uid} #{@options['dest']}" if @options['sync_userid'] == 'from_host'
```

Comment:
You maybe didn't have remarked that the `chown` is not recursive. The only permission issue that we need to manage manually is the 'base' folder. `unison` handles the permission of all files / directories created inside the 'base' folder with the use of  `user`, `group` and `numericids` options.